src/components/button/button.component.js

```diff
@@ -41,97 +98,46 @@ const renderStyledButton = (buttonProps) => {
 
   return (
     <StyledButton
-      as={!disabled && href ? "a" : "button"} // legacy link button feature
+      as={!disabled && href ? "a" : "button"}
+      onKeyDown={href && handleLinkKeyDown}
+      draggable={false}
       buttonType={buttonType}
       disabled={disabled}
       role="button"
-      type="button"
+      type={href ? undefined : "button"}
       iconType={iconType}
       size={size}
       px={px || paddingX}
       noWrap={noWrap}
       {...tagComponent("button", buttonProps)}
-      {...styleProps}
-      ref={forwardRef}
+      {...rest}
+      ref={ref}
     >
       {renderChildren(buttonProps)}
     </StyledButton>
   );
 };
 
 const Button = (props) => {
-  const { disabled, to, iconType, renderRouterLink, size, subtext } = props;
+  const { size, subtext } = props;
+
+  const linkRef = useRef(null);
 
   const { as, buttonType, ...rest } = props;
 
   const propsWithoutAs = {
     ...rest,
     buttonType: buttonType || as,
+    linkRef,
```

Comment:
We can simplify `renderStyledButton` if we move the conditional into this function.

```js
  const { as, buttonType, forwardRef ...rest } = props;
  const propsWithoutAs = {
    ...rest,
    buttonType: buttonType || as,
    ref: forwardRef || ref
  };
```

Reply:
Changed!